remote/remote.go

```diff
@@ -194,11 +194,13 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
 		if err != nil {
 			if err == io.EOF && i != maxRetries {
-				continue // retry if EOF
+				continue // retry
 			}
-			if transportErr, ok := err.(*transport.Error); ok && len(transportErr.Errors) > 0 {
-				switch transportErr.StatusCode {
-				case http.StatusNotFound, http.StatusUnauthorized:
+			if transportErr, ok := err.(*transport.Error); ok && is40x(transportErr) {
+				if i != maxRetries {
```

Comment:
What is the cost of retrying? If an image is truly not on a registry, what kind of impact are we introducing to that situation?

Reply:
If I am not wrong @natalieparellano estimated that the following [fix](https://github.com/buildpacks/imgutil/pull/117)    was saving us 300 milliseconds on each retry, we are retrying 3 times I believe the impact is about 900 milliseconds overhead if the image actually doesn't exist in the registry

Reply:
Are we concerned about that? I feel like I want to be able to send something in like `WithExpectedToExist()` 😄  so that the consumer of this function can add on the behavior. In the case of an app image in analyzer, for instance, I wouldn't want to retry on 404s since it is likely this is a first build situation. But for fetching a run image, I would expect it to exist.